### PR TITLE
Group convolution backprop data

### PR DESCRIPTION
**Pull request by jdanieck**
GroupConvolutionBackpropData reference implementation developed. Ticket 37444.

Implementation details:
- Reused runtime::reference::convolution_backprop_in() for actual operation calculation
- Removed GroupConvolutionBackpropData from int_executable
- Added GroupConvolutionBackpropData to evaluates_map
- SLT tests extended by 1-D GroupConvolutionBackpropData tests
- Serialization SLT added
- Specification refactored
- Ngraph operation refactored

### CODE REVIEW

File: inference-engine/tests/functional/shared_test_classes/src/single_layer/group_convolution.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright (C) 2020 Intel Corporation
+// Copyright (C) 2021 Intel Corporation
```
Comment (iefode) [inline inference-engine/tests/functional/shared_test_classes/src/single_layer/group_convolution.cpp:1]: 2020-2021

Comment (jdanieck) [reply inference-engine/tests/functional/shared_test_classes/src/single_layer/group_convolution.cpp]: Fixed.

**Review by ilyachur — APPROVED** [review]

**Comment by jdanieck** [comment]
@lazarevevgeny following the convention from previous PRs please look at the spec - it is written by me, I followed already established conventions. Ilya already approved, so we're waiting for you and then we can merge.

**Review by lazarevevgeny — APPROVED** [review]

**Review by iefode — APPROVED** [review]
IE Test part LGTM

**Pull request merged**